From 8616b356faa16beb00ca14832c12b1824b248165 Mon Sep 17 00:00:00 2001
From: chenjh <chenjh@rock-chips.com>
Date: Sat, 27 May 2017 11:30:29 +0800
Subject: [PATCH] firmware: rockchip: use sp_el1 from bl31 delivers

we think 'if (fiq_pt_regs.pstate & 0x10)' doesn't make any
sense, use sp_el1 from bl31 delivers is ok.

Change-Id: I0792d76e39912b4ca5484b029761daac05cd719b
Signed-off-by: chenjh <chenjh@rock-chips.com>
---
 drivers/firmware/rockchip_sip.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/firmware/rockchip_sip.c b/drivers/firmware/rockchip_sip.c
index f813dc7048f2..df4cd460ad2c 100644
--- a/drivers/firmware/rockchip_sip.c
+++ b/drivers/firmware/rockchip_sip.c
@@ -144,13 +144,7 @@ static struct pt_regs sip_fiq_debugger_get_pt_regs(void *reg_base,
 
 	/* copy pstate */
 	memcpy(&fiq_pt_regs.pstate, reg_base + 0x110, 8);
-
-	/* EL1 mode */
-	if (fiq_pt_regs.pstate & 0x10)
-		memcpy(&fiq_pt_regs.sp, reg_base + 0xf8, 8);
-	/* EL0 mode */
-	else
-		fiq_pt_regs.sp = sp_el1;
+	fiq_pt_regs.sp = sp_el1;
 
 	/* copy pc */
 	memcpy(&fiq_pt_regs.pc, reg_base + 0x118, 8);
-- 
2.35.3

